src/components/Popups/TransactionPopup.tsx

```diff
@@ -15,21 +19,32 @@ const RowNoFlex = styled(AutoRow)`
   margin-right: 16px;
 `
 
-export function TransactionPopup({ hash, success, summary }: PopupContent) {
+export function TransactionPopup({ hash, success, summary, swapProtocol }: PopupContent) {
   const { chainId } = useActiveWeb3React()
 
+  const { t } = useTranslation()
   const theme = useTheme()
 
+  const isGnosisProtocolHash = swapProtocol === SwapProtocol.COW
+
+  const explorer = isGnosisProtocolHash ? t('gnosisProtocolExplorer') : t('blockExplorer')
+
+  const link = isGnosisProtocolHash
+    ? getGnosisProtocolExplorerOrderLink(chainId as ChainId, hash)
+    : getExplorerLink(chainId as ChainId, hash, 'transaction')
+
+  const popupText = `${t('viewOn')} ${explorer}`
+
+  const explorerExternalLink = <ExternalLink href={link}>{popupText}</ExternalLink>
+
   return (
     <RowNoFlex>
       <div style={{ paddingRight: 16 }}>
         {success ? <CheckCircle color={theme.green1} size={24} /> : <AlertCircle color={theme.red1} size={24} />}
       </div>
       <AutoColumn gap="8px">
         <TYPE.body fontWeight={500}>{summary ?? 'Hash: ' + hash.slice(0, 8) + '...' + hash.slice(58, 65)}</TYPE.body>
-        {chainId && (
-          <ExternalLink href={getExplorerLink(chainId, hash, 'transaction')}>View on block explorer</ExternalLink>
-        )}
+        {explorerExternalLink}
```

Comment:
`explorerExternalLink` is just one line jsx without any sophisticated logic, wouldn't be better just to put `<ExternalLink href={link}>{popupText}</ExternalLink>` here?

Reply:
fair. done